### ISSUE TITLE
ElementHelper::createSlug and limitAutoSlugsToAscii config

### ISSUE THREAD

**Issue by iliaivanov** [craftcms/cms#2427]
### Description
`ElementHelper::createSlug` helper method doesn't take into account `limitAutoSlugsToAscii` configuration. I think it's a confusing a bit because I personally expected that behavior will be the same as for saving entries from UI.

### Additional info
- Craft version: Craft CMS 3.0.0-RC7.1

**Comment by brandonkelly** [comment]
It’s enforced in SlugValidator: https://github.com/craftcms/cms/blob/ad740287644ea113206212e55046bbeb16b64a47/src/validators/SlugValidator.php#L68-L70

But note the setting is not called `limitSlugsToAscii` – the `Auto` means it will only be taken into account if there was no user-defined slug to begin with. So if you submit an entry without any slug, one will be generated based on the entry’s title, and that’s when the setting will come into play.

**Comment by iliaivanov** [comment]
Yeah, I see and I ended up at this validator while was riding code with xdebug. But imagine that for a reason I need to change slug from code. I register a listener for the `Elements::EVENT_BEFORE_SAVE_ELEMENT` event and callback function is responsible of creating new slug value for the entry. This means that using `ElementHelper::createSlug` and assigning this value as the entry slug wouldn't take `limitAutoSlugsToAscii` setting into account and `SlugValidator` wouldn't convert it to the AscII either because slug value is already defined.

**Comment by brandonkelly** [comment]
Yeah I see your point but `ElementHelper::createSlug()` assumes that whatever was passed in was user-defined, so it’s more concerned with ensuring the slug is valid, not that it respects auto-generation rules.

Your event listener could just provide its own support for the config setting:

```php
if (Craft::$app->getConfig()->getGeneral()->limitAutoSlugsToAscii) {
    $slug = StringHelper::toAscii($slug);
}
```

I think the only way we would make `ElementHelper::createSlug()` care about converting to ASCII is if we renamed the config setting to `limitSlugsToAscii` (dropping the `Auto`), in which case it stops mattering whether it was user-defined or not. Feel free to request that in a separate issue, but we would need to get more people involved in that discussion before we made a chance like that.
